hometask2.js

```diff
@@ -0,0 +1,130 @@
+/*
+Get the Middle Character
+You are going to be given a word. Your job is to return the middle character of the word. If the word's length is odd, return the middle character. If the word's length is even, return the middle 2 characters.
+
+#Examples:
+
+Kata.getMiddle("test") should return "es"
+
+Kata.getMiddle("testing") should return "t"
+
+Kata.getMiddle("middle") should return "dd"
+
+Kata.getMiddle("A") should return "A"
+#Input
+
+A word (string) of length 0 < str < 1000 (In javascript you may get slightly more than 1000 in some test cases due to an error in the test cases). You do not need to test for this. This is only here to tell you that you do not need to worry about your solution timing out.
+
+#Output
+
+The middle character(s) of the word represented as a string.
+*/
+function getMiddle(s) {
+    const stringLength = s.length;
+    const middle = parseInt(stringLength / 2);
+    if (stringLength % 2 != 0) {
+        return s[middle];
+    }
+    return s[middle - 1] + s[middle];
+}
+
+/*
+Partition On
+DESCRIPTION:
+Write a function which partitions a list of items based on a given predicate.
+
+After the partition function is run, the list should be of the form [ F, F, F, T, T, T ] where the Fs (resp. Ts) are items for which the predicate function returned false (resp. true).
+
+NOTE: the partitioning should be stable; in other words: the ordering of the Fs (resp. Ts) should be preserved relative to each other.
+
+For convenience and utility, the partition function should return the boundary index. In other words: the index of the first T value in items.
+
+For example:
+var items = [1, 2, 3, 4, 5, 6];
+function isEven(n) {return n % 2 == 0}
+var i = partitionOn(isEven, items);
+// items should now be [1, 3, 5, 2, 4, 6]
+// i     should now be 3
+
+*/
+function partitionOn(pred, items) {
+    let truePredicate = items.filter(item => pred(item));
+    let falsePredicate = items.filter(item => !pred(item));
+    let i = 0;
+    for (let item of falsePredicate) {
```

Comment:
Make an habit to not modify function argument and no need to use create these two separate for loops use concat or spread operator